### PR TITLE
Expansion Hunter module, #442

### PR DESCRIPTION
**Pull request by jemten**

### CODE REVIEW

File: lib/MIP/Program/Variantcalling/Expansionhunter.pm [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+## Function : Perl wrapper for writing Expansion Hunter command to $FILEHANDLE or return commands array.
+## Returns  : @commands
+## Arguments: $FILEHANDLE              => Filehandle to write to
+##          : $infile_path             => Path to sorted, indexed bam file
+##          : $json_outfile_path       => Path to json file
+##          : $log_outfile_path        => Path to log file
+##          : $min_anchor_mapq         => Min MAPQ of an in-repeat read anchor
+##          : $min_baseq               => Min base quality of a high confidece base call
+##          : $min_score               => Min weigheted matching score
+##          : $read_depth              => Expansion Hunter will not calciulate read depth if set
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Expansionhunter.pm]: calciulate -> calculate

File: lib/MIP/Program/Variantcalling/Expansionhunter.pm [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+            stderrfile_path        => $stderrfile_path,
+            stderrfile_path_append => $stderrfile_path_append,
+            stdoutfile_path        => $stdoutfile_path,
+        }
+      );
+
+    unix_write_to_file(
+        {
+            FILEHANDLE   => $FILEHANDLE,
+            commands_ref => \@commands,
```
Comment (henrikstranneheim) [inline lib/MIP/Program/Variantcalling/Expansionhunter.pm]: Order AB

File: lib/MIP/Recipes/Pipeline/Rare_disease.pm [hunk 3]
```diff
@@ -838,6 +839,41 @@ sub pipeline_rare_disease {
...
+
+            my $outsample_directory = catdir(
+                $active_parameter_href->{outdata_dir},
+                $sample_id, $active_parameter_href->{outaligner_dir},
+                $cnvnator_program_name
+            );
+
+            analysis_expansionhunter(
+                {
+                    parameter_href          => $parameter_href,
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Rare_disease.pm]: Order AB

File: lib/MIP/Recipes/Pipeline/Rare_disease.pm [hunk 4]
```diff
@@ -838,6 +839,41 @@ sub pipeline_rare_disease {
             }
         );
     }
+    if ( $active_parameter_href->{pexpansionhunter} ) {
+
+        $log->info(q{[ExpansionHunter]});
+
+        my $cnvnator_program_name = q{expansionhunter};
+
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Rare_disease.pm]: Add loop label

**Comment by jemten** [comment]
Ooos, the pipeline recipe wasn't supposed to be a part of this pull request. (To eager with `git add -u`). I will change according to your suggestions and possibly update it again when it is time for the analysis recipe.

**Comment by henrikstranneheim** [comment]
I wondered about that. Good plan!

**Comment by henrikstranneheim** [comment]
Alright so I will not merge this until you say so!

**Comment by jemten** [comment]
I think it is OK to merge it now. Since $active_parameter_href->{pexpansionhunter} isn't defined anywhere yet it shouldn't interfere with anyone else's work. Would be nice to get this part out of the way.

**Pull request merged**